crates/ide-assists/src/handlers/inline_call.rs

```diff
@@ -358,16 +358,15 @@ fn inline(
                 .first_token()
                 .expect("NameRef should have had a token.")
         };
-        if let Some(self_local) = params[0].2.as_local(sema.db) {
-            usages_for_locals(self_local)
-                .filter_map(|FileReference { name, range, .. }| match name {
-                    FileReferenceNode::NameRef(_) => Some(body.syntax().covering_element(range)),
-                    _ => None,
-                })
-                .for_each(|usage| {
-                    ted::replace(usage, &this());
-                });
-        }
+
+        body.syntax()
+            .descendants_with_tokens()
+            .filter_map(NodeOrToken::into_token)
+            .filter(|tok| tok.kind() == SyntaxKind::SELF_KW)
```

Comment:
That approach won't work, as `self` can also refer to the current module

Reply:
You are right, I forgot the other semantics of `self` 😱